src/unix/fs.c

```diff
@@ -217,6 +217,88 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  int r;
+#ifdef UV__O_CLOEXEC
+  int (*mkostemp_function)(char*, int);
+  static int no_cloexec_support;
+#endif
+  char *path;
+  int path_length;
+
+  path = (char*) req->path;
+  path_length = strlen(path);
+
+  /* EINVAL can be returned for 2 reasons:
+       1. The template's last 6 characters were not XXXXXX
+       2. open() didn't support O_CLOEXEC
+     We want to avoid going to the fallback path in case
+     of 1., so it's manually checked before. */
+  if (path_length < 6 ||
+      strcmp(path + (path_length - 6), "XXXXXX")) {
+    errno = EINVAL;
+    return -1;
+  }
+
+#ifdef UV__O_CLOEXEC
+  if (no_cloexec_support == 0) {
+    *(int**)(&mkostemp_function) = dlsym(RTLD_DEFAULT, "mkostemp64");
+
+    /* We don't care about errors, but we do want to clean them up.
+       If there has been no error, then dlerror() will just return
+       NULL. */
+    dlerror();
+
+    /* If mkostemp64() doesn't exist, try mkostemp(). */
+    if (mkostemp_function == NULL) {
+      *(int**)(&mkostemp_function) = dlsym(RTLD_DEFAULT, "mkostemp");
+      dlerror();
+    }
+
+    if (mkostemp_function != NULL) {
+      r = mkostemp_function(path, UV__O_CLOEXEC | O_LARGEFILE);
+
+      if (r >= 0)
+        return r;
+
+      /* If mkostemp() returns EINVAL, it means the kernel doesn't
+         support O_CLOEXEC, so we just fallback to mkstemp() below. */
+      if (r == -1 && errno != -EINVAL)
```

Comment:
Sorry, that's what I mean; the important part was not checking the negated version. :-)